### PR TITLE
Update KRO Class-Shadow costume

### PR DESCRIPTION
**Pull request by attackjom**
* **Addressed Issue(s)**:  -

* **Server Mode**:  Renewal

* **Description of Pull Request**: 

- Fix SkillUseSPrate script in Class1-S.costume
- Add Class2 & Class3 & Doram & Extend Class - S.Costume

Thank to "Sigma" for the translation in divine-pride forum
https://www.divine-pride.net/forum/index.php?/topic/2998-archive-classes-shadow-set/

### CODE REVIEW

**Comment by Atemo** [comment]
Do you happen to have the data for the IDs between 24319 and 24402? Coincidentally I started to work on them some days ago but if you have them already it would be a waste.

**Comment by attackjom** [comment]
Nope I just have the class-shadow costume only

File: db/re/item_db.txt [hunk 1]
```diff
@@ -11997,17 +11997,95 @@
...
-24254,S_Magician_Pendant,Magician Shadow Pendant,12,0,,0,,,,0,0xFFFFFFFF,63,2,2097152,,1,1,,{ .@r = getrefine(); bonus2 bMagicAtkEle,Ele_Fire,20+(5*.@r); },{},{}
+24254,S_Magician_Pendant,Magician Shadow Pendant,12,0,,0,,,,0,0xFFFFFFFF,63,2,2097152,,1,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"MG_FIREWALL",20+5*.@r; },{},{}
 24255,S_Archer_Pendant,Archer Shadow Pendant,12,0,,0,,,,0,0xFFFFFFFF,63,2,2097152,,1,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"AC_SHOWER",20+(5*.@r); },{},{}
+24256,Knight_Shadow_Shoes,Knight Shadow Shoes,12,,,0,,,,,0xFFFFFFFF,63,2,524288,,1,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"KN_PIERCE",20+.@r*5; },{},{}
+24257,Crusader_Shadow_Shoes,Crusader Shadow Shoes,12,,,0,,,,,0xFFFFFFFF,63,2,524288,,1,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"CR_HOLYCROSS",20; bonus bUseSPrate,-3*.@r; },{},{}
+24258,Blacksmith_Shadow_Shoes,Blacksmith Shadow Shoes,12,,,0,,,,,0xFFFFFFFF,63,2,524288,,1,1,,{ .@r = getrefine(); bonus2 bSkillUseSPrate,"BS_HAMMERFALL",20; bonus2 bSkillUseSPrate,"BS_ADRENALINE",20; bonus bUseSPrate,-5*.@r; },{},{}
```
[the remark this answers is not in the document]
Comment (attackjom) [reply db/re/item_db.txt]: bonus2 bSkillUseSPrate,sk,n;		Decreases SP consumption of skill sk by n%

**Comment by attackjom** [comment]
@Atemo  Thanks you for the huge correction of my mistake.
I have already fix it to your suggestion.

File: db/re/item_combo_db.txt [hunk 2]
```diff
@@ -197,12 +197,63 @@
...
+24286:24317,{ bonus bMaxHPrate,2; bonus bMaxSPrate,2; bonus bAllStats,3; bonus bUseSPrate,-10; }
+24287:24316,{ bonus bMaxHPrate,2; bonus bMaxSPrate,2; bonus bAllStats,3; bonus bUseSPrate,-10; }
+24288:24301,{ bonus2 bSkillUseSPrate,"RK_SONICWAVE",20; bonus3 bAutoSpell,"RK_IGNITIONBREAK",3,20; }
+24288:24301:24246:24250:24256:24269,{ bonus bAllStats,10; bonus bMaxHPrate,5; bonus bMaxSPrate,5; bonus2 bIgnoreDefClassRate,Class_Normal,50; bonus2 bIgnoreMDefClassRate,Class_Normal,50; }
+24289:24302,{ bonus2 bSkillUseSPrate,"LG_CANNONSPEAR",20; skill "AM_CP_SHIELD",3; }
+24289:24302:24246:24250:24257:24270,{ bonus bAllStats,10; bonus bMaxHPrate,5; bonus bMaxSPrate,5; bonus2 bIgnoreDefClassRate,Class_Normal,50; bonus2 bIgnoreMDefClassRate,Class_Normal,50; }
+24290:24303,{ bonus2 bSkillAtk,"NC_VULCANARM",20; bonus2 bSkillUseSPrate,"NC_FLAMELAUNCHER",20; }
+24291:24304,{ bonus bMaxHPrate,2; bonus2 bSkillCooldown,"GN_CART_TORNADO",-1000; bonus2 bVariableCastrate,"GN_CARTBOOST",-100; }
+24292:24305,{ bonus2 bSkillAtk,"AB_DUPLELIGHT",20; bonus2 bSkillAtk,"AB_DUPLELIGHT_MELEE",20; bonus2 bSkillAtk,"AB_DUPLELIGHT_MAGIC",20; bonus2 bSkillCooldown,"AB_ADORAMUS",-1000; }
```
Comment (Atemo) [inline db/re/item_combo_db.txt]:
```suggestion
24292:24305,{ bonus2 bSkillAtk,"AB_DUPLELIGHT",20; bonus2 bSkillCooldown,"AB_ADORAMUS",-1000; }
```

File: db/re/item_db.txt [hunk 3]
```diff
@@ -11997,17 +11997,95 @@
...
+24288,Rune_Knight_Shadow_Weapon,Rune Knight Shadow Weapon,12,10,,0,,0,,0,0x00000080,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"RK_SONICWAVE",20+.@r*5; },{},{}
+24289,Royal_Guard_Shadow_Weapon,Royal Guard Shadow Weapon,12,10,,0,,0,,0,0x00004000,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"LG_CANNONSPEAR",20+.@r*5; },{},{}
+24290,Mechanic_Shadow_Weapon,Mechanic Shadow Weapon,12,10,,0,,0,,0,0x00000400,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"NC_VULCANARM",20+.@r*5; },{},{}
+24291,Genetic_Shadow_Weapon,Genetic Shadow Weapon,12,10,,0,,0,,0,0x00040000,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"GN_CART_TORNADO",20+.@r*5; },{},{}
+24292,Arcbishop_Shadow_Weapon,Arcbishop Shadow Weapon,12,10,,0,,0,,0,0x00000100,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"AB_DUPLELIGHT",20+.@r*5; bonus2 bSkillAtk,"AB_DUPLELIGHT_MELEE",20+.@r*5; bonus2 bSkillAtk,"AB_DUPLELIGHT_MAGIC",20+.@r*5; },{},{}
```
Comment (Atemo) [inline db/re/item_db.txt]:
```suggestion
24292,Arcbishop_Shadow_Weapon,Arcbishop Shadow Weapon,12,10,,0,,0,,0,0x00000100,56,2,131072,,99,1,,{ .@r = getrefine(); bonus2 bSkillAtk,"AB_DUPLELIGHT",20+.@r*5; },{},{}
```

Comment (attackjom) [reply db/re/item_db.txt]: done

File: db/re/item_combo_db.txt (hunk 2 above)
Comment (attackjom) [reply db/re/item_combo_db.txt]: done

**Comment by attackjom** [comment]
Nope let’s go

**Comment by Atemo** [comment]
Thanks @attackjom !

**Pull request merged**